llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2823,7 +2829,7 @@ void VPlanTransforms::narrowInterleaveGroups(VPlan &Plan, ElementCount VF,
     auto *S = new VPWidenStoreRecipe(
         *cast<StoreInst>(StoreGroup->getInterleaveGroup()->getInsertPos()),
         StoreGroup->getAddr(), Res, nullptr, /*Consecutive=*/true,
-        /*Reverse=*/false, StoreGroup->getDebugLoc());
+        /*Reverse=*/false, {}, StoreGroup->getDebugLoc());
```

Comment:
```suggestion
        /*Reverse=*/false, VPIRMetadata(SI), StoreGroup->getDebugLoc());
```
similar to LI above?

Reply:
This load combines all scalar loads from the group into a single wide store; we would have to use the intersection of metadata on all stores. I updated forming the load as well and added a test showing the issue in 127f48668b719798770c36a20288ef5230669f42.

Also dropped NFC from the title